.github/actions/wait-for-job-succeed-or-fail/action.yaml

```diff
@@ -0,0 +1,32 @@
+# Waits for a given job to finish
+# On succees it passes, on failure it fails the job
+# inputs: job-name (required), timeout (optional)
+# Don't forget to use actions/checkout@v3 and actions/setup-go@v4 before
```

Comment:
I think it should be possible to move the setup-go action also inside of the composite actions as step? This way the actions would be more self-contained.

```
runs:
  using: "composite"
  steps:
    - uses: actions/setup-go@v4
       with:
         go-version: "1.20"
    - name: Wait for job to succeed
```